### PR TITLE
Fixes #321 Add option to remove repository from Yaydoc

### LINKED ISSUE #321: Add option to completely remove a repository from Yaydoc
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

A user can only register a repository to Yaydoc and there is no option to remove the repository.
### Expected behaviour

The user should be allowed to completely delete the repository from Yaydoc, including the Webhook

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

* Delete Repository from database and also remove the webhook
* Add modal to add confirmation for deleting repository.

## Related Issue

issue #321 
## How Has This Been Tested?

https://yaydoc6.herokuapp.com
## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296  @sch00lb0y  Please review.

File: backend/github.js [hunk 1]
```diff
@@ -53,3 +53,26 @@ exports.retrieveOrganizations = function (accessToken, callback) {
...
 };
+
+/**
+ * Delete a Repository WebHook
+ * @param name: name of the Repository
+ * @param hook: Hook Id
+ * @param accessToken: Access Token of the user
+ * @param callback
+ */
+exports.deleteHook = function (name, hook, accessToken, callback) {
```
Comment (sch00lb0y) [inline backend/github.js]: can we shift to Promises from callback? consistency will be followed everywhere. we can't avoid the use of Promises but then we can avoid the use of the callback.

Comment (imujjwal96) [inline backend/github.js]: There is more code written using callbacks than using Promises, updating the structure of Yaydoc would require a lot of time. Using callbacks is more of a readability problem than usages problem.
The code changes could be done at later stages, if required.

File: routes/repository.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  var name = req.body.name || '';
+
+  if (name === '') {
+    return res.redirect('/dashboard?status=delete_failure');
+  }
+
+  async.waterfall([
+    function (callback) {
+      Repository.getRepositoryByName(name, function (error, repository) {
+        callback(null, repository);
```
Comment (sch00lb0y) [inline routes/repository.js]: any reason for not handling this error?

Comment (imujjwal96) [inline routes/repository.js]: Updated!

**Comment by imujjwal96** [comment]
@pri22296  @sch00lb0y  Please review.

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Please review.

**Comment by pri22296** [comment]
Application error on sign in to dashboard

**Comment by imujjwal96** [comment]
@pri22296  Check again

**Comment by pri22296** [comment]
Worked this time. I registered a repo. then deleted it. worked fine. But then registering again, I got a `registration mismatch`

**Comment by imujjwal96** [comment]
@pri22296 Should be fixed now!

**Comment by pri22296** [comment]
Again getting application error

**Comment by imujjwal96** [comment]
@pri22296 I'll have to dig deeper on that bug. I understand that it is undesirable and it seems to happen everytime i redeploy to heroku

**Comment by pri22296** [comment]
Anyways when does `registration_mismatch` happen? Just asking for info.

**Comment by imujjwal96** [comment]
When a hook on the repository exists, bur the repository doesn't exist in our DB. It'll happen when the repository is manually deleted from the database

**Comment by pri22296** [comment]
@imujjwal96 Please resolve conflicts

**Comment by imujjwal96** [comment]
@pri22296 Resolved.

**Comment by pri22296** [comment]
@imujjwal96 Please make the following changes.

- The confirmation modal should have the focus when it's opened. Currently, I need to click on it before beginning to type.
- Also pressing return should also act as confirmation instead of having to press delete.

**Comment by imujjwal96** [comment]
@pri22296 Updated it. 
The submit on return functionality was already working on my end. Can you check that again?

**Comment by pri22296** [comment]
Yep, worked this time. should have been some issue from my end.

**Pull request merged**